### PR TITLE
Support for Bazel buildsystem

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

File: examples/standalone/window/window.cc [hunk 1]
```diff
@@ -34,8 +36,19 @@ int main(int _argc, char **_argv)
...
+      "<ambient_light>1.0 0 0</ambient_light>"
+      "<background_color>0 1 0</background_color>"
+      "<camera_pose>1 2 3 0 0 1.57</camera_pose>"
+    "</plugin>";
+  tinyxml2::XMLDocument pluginDoc;
+  pluginDoc.Parse(pluginStr);
+
   // Load plugins / config
-  if (!app.LoadPlugin("Publisher"))
+  if (!app.LoadPlugin("Scene3D", pluginDoc.FirstChildElement("plugin")))
```
Comment (chapulina) [inline examples/standalone/window/window.cc:51]: Is this change needed? This being an introductory example, it's preferable to use a simple plugin like `Publisher` instead of a complex one like `Scene3D`. Maybe we can add another example?

File: src/Application_TEST.cc [hunk 2]
```diff
@@ -84,16 +84,17 @@ TEST(ApplicationTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LoadPlugin))
...
     app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
 
     EXPECT_TRUE(app.LoadPlugin("TestPlugin"));
   }
 
   // Plugin path added by env var
   {
-    setenv("TEST_ENV_VAR",
-        (std::string(PROJECT_BINARY_PATH) + "/lib").c_str(), 1);
+    auto env_var = std::string(PROJECT_BINARY_PATH);
```
Comment (chapulina) [inline src/Application_TEST.cc:95]:
```suggestion
    auto envVar = std::string(PROJECT_BINARY_PATH);
```

File: src/Application.cc [hunk 3]
```diff
@@ -306,6 +306,7 @@ bool Application::LoadPlugin(const std::string &_filename,
 
   common::SystemPaths systemPaths;
   systemPaths.SetPluginPathEnv(this->dataPtr->pluginPathEnv);
+  systemPaths.AddPluginPaths("./" + std::string(IGN_GUI_PLUGIN_PATH));
```
Comment (chapulina) [inline src/Application.cc:309]: Can you explain what's the intention here? `IGN_GUI_PLUGIN_PATH` doesn't exist and this fails to compile. This also doesn't seem to be platform-independent.

File: src/Application.cc [hunk 4]
```diff
@@ -570,7 +570,8 @@ std::vector<std::pair<std::string, std::vector<std::string>>>
   std::string home;
   common::env(IGN_HOMEDIR, home);
   paths.push_back(home + "/.ignition/gui/plugins");
-
+  paths.push_back(common::cwd() + "/" + std::string(IGN_GUI_PLUGIN_PATH));
```
Comment (chapulina) [inline src/Application.cc:573]: I'm not sure what the intention is here, but the plugins installed with `ign-gui` are meant to be covered by the case below. Isn't Bazel installing them to `IGN_GUI_PLUGIN_INSTALL_DIR`?

Another thing to consider when adding plugin paths is that Ignition GUI will display all shared libraries it finds on those paths under the top-right menu. So we don't want to be adding paths where non-GUI plugins can also be found.

File: src/Application_TEST.cc [hunk 5]
```diff
@@ -84,16 +84,17 @@ TEST(ApplicationTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LoadPlugin))
   // Plugin path added programmatically
   {
     Application app(g_argc, g_argv);
+    app.AddPluginPath(std::string(PROJECT_BINARY_PATH));
```
Comment (chapulina) [inline src/Application_TEST.cc:87]: Is this necessary because Bazel isn't installing the plugins to `lib`? I have no idea of what's involved in this, but wouldn't it be preferable if the Bazel setup would install libraries there?

File: src/Plugin.cc [hunk 6]
```diff
@@ -124,9 +124,21 @@ void Plugin::Load(const tinyxml2::XMLElement *_pluginElem)
   this->dataPtr->context->setContextProperty(QString::fromStdString(filename),
       this);
 
+  // TODO perhaps find a more generic way of locating these resource files
```
Comment (chapulina) [inline src/Plugin.cc:127]: Yeah we need  to do that, using build-defined paths (i.e. not `cwd`) and making sure it is cross-platform.

File: src/cmd/cmdgui.rb.in [hunk 7]
```diff
@@ -28,7 +28,7 @@ end
 require 'optparse'
 
 # Constants.
-LIBRARY_NAME = '@IGN_LIBRARY_NAME@'
+LIBRARY_NAME = '@library_location@'
```
Comment (chapulina) [inline src/cmd/cmdgui.rb.in:31]: Does this work with cmake?

**Review by chapulina — CHANGES_REQUESTED** [review]
Did a quick pass. Also note that there are codecheck complaints.

File: BUILD.bazel [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+        "IGN_GUI_STATIC=true",
+    ],
+    includes = ["include"],
+    deps = [
+        "@qt//:qt_core",
+        "@qt//:qt_gui",
+        "@qt//:qt_network",
+        "@qt//:qt_qml",
+        "@qt//:qt_quick",
+        "@qt//:qt_quickcontrols2",
```
Comment (JShep1) [inline BUILD.bazel:146]:
```suggestion
        "@qt//:qt_quick_control",
```

**Review by JShep1 — COMMENTED** [review]
I'm getting errors for multiple definitions of `ignstrtok` which is defined in `ign_msgs` and `ign_common`

File: BUILD.bazel (hunk 8 above)
Comment (mahiuchun) [reply BUILD.bazel:146]: +1

**Review by chapulina — CHANGES_REQUESTED** [review]
Besides all my other comments, CI is also broken

**Comment by chapulina** [comment]
I believe this PR is outdated and not being worked on. Please reopen if this starts being worked on again.
